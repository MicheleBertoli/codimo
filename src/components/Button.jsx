/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import { BUTTON_COLORS } from 'constants/colors';

const AnimatedButton = styled.a`
  border-radius: 10px;
  cursor: ${({ disabled }) => (disabled ? 'cursor' : 'pointer')};
  float: left;
  font-family: 'Allerta', sans-serif;
  font-size: 25px;
  font-weight: bold;
  opacity: ${({ disabled }) => (disabled ? '0.6' : 'auto')};
  padding: 10px 40px;
  pointer-events: ${({ disabled }) => (disabled ? 'none' : 'auto')};
  text-decoration: none;
  transition: all 0.1s;
  &:active {
    border-bottom: 1px solid;
    transform: translate(0, 5px);
  }
`;
const ColoredAnimatedButton = AnimatedButton.extend`
  background-color: ${({ type }) => BUTTON_COLORS[type].bgColor};
  border-bottom: 5px solid ${({ type }) => BUTTON_COLORS[type].border};
  color: ${({ type }) => BUTTON_COLORS[type].color};
  text-shadow: 0 -2px ${({ type }) => BUTTON_COLORS[type].border};
`;

type Props = {|
  disabled?: boolean,
  title: string,
  type: $Keys<typeof BUTTON_COLORS>,
  handleClick: (event: SyntheticEvent) => void,
|};
export default function ExecuteButton({ handleClick, title, type, disabled }: Props) {
  return (
    <ColoredAnimatedButton
      type={type}
      disabled={disabled}
      onClick={handleClick}
    >{title}</ColoredAnimatedButton>
  );
}
