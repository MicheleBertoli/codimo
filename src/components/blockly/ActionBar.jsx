/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 */
import React from 'react';
import styled from 'styled-components';

import Button from './Button';

const Bar = styled.div`
  align-items: center;
  background-color: #e3e3e3;
  border: 1px solid #ddd;
  border-bottom: 0;
  display: flex;
  height: 68px;
  width: 628px;
  & * {
    margin: 0 0 0 10px;
  }
`;

type Props = {|
  isExecuting: boolean,
  isStopped: boolean,
  handleStartGame(): void,
  handleResetGame(): void,
|};
// FIXME hardcoded title
const ActionBar = ({ isExecuting, isStopped, handleStartGame, handleResetGame }: Props) => (
  <Bar>
    {isStopped ?
      <Button
        title="▶ Dale play!"
        type="green"
        handleClick={handleStartGame}
      /> :
      <Button
        disabled={isExecuting}
        title="⏹ Reseteá!"
        type="orange"
        handleClick={handleResetGame}
      />
    }
  </Bar>
);

export default ActionBar;
