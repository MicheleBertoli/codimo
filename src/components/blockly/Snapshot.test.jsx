/**
 * @author Luciano Graziani @lgraziani2712
 * @license {@link http://www.opensource.org/licenses/mit-license.php|MIT License}
 *
 * @flow
 * @jest-environment node
 */
import React from 'react';
import renderer from 'react-test-renderer';

import ActionBar from './ActionBar';
import Button from './Button';

describe('Components snapshots', () => {
  it('renders the Button correctly', () => {
    const buttonTree = renderer.create(
      <Button handleClick={() => {}} title="I'm a button" type="red" />,
    ).toJSON();

    expect(buttonTree).toMatchSnapshot();
  });
  it('renders the ActionBar correctly', () => {
    const tree = renderer.create(
      <ActionBar
        isExecuting={false}
        isStopped={true}
        handleStartGame={() => {}}
        handleResetGame={() => {}}
      />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
