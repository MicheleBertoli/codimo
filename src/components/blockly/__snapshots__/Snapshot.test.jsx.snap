// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components snapshots renders the ActionBar correctly 1`] = `
<div
  className="ActionBar__Bar-fFHfuW eSZxqI"
>
  <a
    className="Button__ColoredAnimatedButton-fylGiI fZVUpp Button__AnimatedButton-jOrBhv dTlQly"
    disabled={undefined}
    onClick={[Function]}
    type="green"
  >
    â–¶ Dale play!
  </a>
</div>
`;

exports[`Components snapshots renders the Button correctly 1`] = `
<a
  className="Button__ColoredAnimatedButton-fylGiI fAibEq Button__AnimatedButton-jOrBhv dTlQly"
  disabled={undefined}
  onClick={[Function]}
  type="red"
>
  I'm a button
</a>
`;
